addons/binding/org.openhab.binding.verisure/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,505 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="verisure"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+
+	<!-- Sample Thing Type -->
+	<bridge-type id="bridge">
+		<label>Verisure Bridge</label>
+		<description>This bridge represents the http://mypages.verisure.com web page connector.</description>
+
+		<channels>
+			<channel id="status" typeId="status" />
+		</channels>
+
+		<properties>
+			<property name="vendor">Verisure</property>
+		</properties>
+
+		<config-description>
+			<parameter name="username" type="text" required="true">
+				<label>Username</label>
+				<context>user</context>
```

Comment:
>There is a lot of code related to sending commands that does look very similar. The code would greatly benefit if you can extract some generic component out of it and pass variables to the method. I think this can greatly reduce the code.

I agree, I will try to use a generic way of sending commands to reduce the amount of code.

Regarding the comment on creating separate handler classes for each device type, I will try to understand how that should be done and what benefits that gives, and then decide how to do.

Reply:
> Regarding the comment on creating separate handler classes for each device type, I will try to understand how that should be done and what benefits that gives, and then decide how to do.

One example that uses this approach is the tplinksmarthome binding (I'm familiar with it because it's a binding I wrote). That also supports several different type of devices. Because the devices are different it becomes easier to see what code belongs to which device and if new devices or changes to a specific device need to be made it's clear where to make changes without worrying about the rest of the code.

Reply:
OK, thanks for the tip :-) I'll look into it!